### PR TITLE
feat(android): support android

### PR DESCRIPTION
**Pull request by sjelin**

### CODE REVIEW

File: lib/cmds/initialize.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+      }
+    });
+  };
+}
+
+// Run a command on the android SDK
+function runAndroidSDKCommand(sdkPath: string, cmd: string, args: string[],
+    spawnOptions: Object, config_fun?: Function): q.Promise<any> {
+
+  var child = child_process.spawn(path.join(sdkPath, 'tools', 'android'),
```
Comment (cnishina) [inline lib/cmds/initialize.ts]: Nit: use `let` over `var`

File: lib/cmds/initialize.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    [cmd].concat(args), spawnOptions);
+
+  if(config_fun) {
+    config_fun(child);
+  };
+
+  var deferred = q.defer()
```
Comment (cnishina) [inline lib/cmds/initialize.ts]: Same nit: use `let` over `var`

File: lib/cmds/update.ts [hunk 3]
```diff
@@ -106,12 +120,30 @@ function update(options: Options): void {
...
+  }
 }
 
-function updateBinary(binary: Binary, outputDir: string, proxy: string, ignoreSSL: boolean) {
-  FileManager.toDownload(binary, outputDir).then((value: boolean) => {
+function updateBinary(binary: Binary, outputDir: string, proxy: string, ignoreSSL: boolean): q.Promise<any> {
+  return FileManager.toDownload(binary, outputDir).then((value: boolean) => {
     if (value) {
-      Downloader.downloadBinary(binary, outputDir, proxy, ignoreSSL, unzip);
+      var deferred = q.defer();
```
Comment (cnishina) [inline lib/cmds/update.ts]: Nit: use `let` over `var`

**Comment by cnishina** [comment]
Could we get some tests in the [fileManager_spec](https://github.com/angular/webdriver-manager/blob/master/spec/files/fileManager_spec.ts)? This will make sure the correct files per OS.

**Comment by cnishina** [comment]
(pressed the wrong button. meant to press "comment" and not "close and comment")

**Comment by sjelin** [comment]
All comments addressed.  Also: docs added

File: package.json [hunk 4]
```diff
@@ -28,6 +28,9 @@
   "dependencies": {
     "adm-zip": "^0.4.7",
     "chalk": "^1.1.1",
+    "del": "^2.2.0",
+    "glob": "^7.0.3",
+    "ini": "^1.3.4",
     "minimist": "^1.2.0",
     "q": "^1.4.1",
     "request": "^2.69.0"
```
Comment (cnishina) [inline package.json]: also needs `"rimraf": "^2.5.2"`

File: lib/cmds/opts.ts [hunk 5]
```diff
@@ -1,38 +1,55 @@
...
+export const AVD_USE_SNAPSHOTS = 'avd-use-snapshots';
 
 /**
  * The options used by the commands.
  */
 var opts: Options = {};
 opts[OUT_DIR] = new Option(OUT_DIR, 'Location to output/expect', 'string', Config.getSeleniumDir());
 opts[SELENIUM_PORT] = new Option(SELENIUM_PORT, 'Optional port for the selenium standalone server', 'string');
+opts[APPIUM_PORT] = new Option(APPIUM_PORT, 'Optional port for the appium server', 'string');
+opts[AVD_PORT] = new Option(AVD_PORT, 'Optional port for android virtual devices.  This port will be used for the console, the port one higher will be used for the ADB.  If mutiple AVDs are being emulated, they will take up the following consecutive ports in the same manner', 'string');
```
Comment (cnishina) [inline lib/cmds/opts.ts]: Could this be shortened? Or possibly reference the mobile readme file?

File: lib/cmds/start.ts [hunk 6]
```diff
@@ -100,6 +107,13 @@ function start(options: Options) {
         '-Dwebdriver.ie.driver=' +
         path.join(outputDir, binaries[IEDriver.id].executableFilename(osType)));
   }
+  if (downloadedBinaries[AndroidSDK.id] != null) {
```
Comment (cnishina) [inline lib/cmds/start.ts]: Could we also hide this behind an `--appium` flag? A user might have the binaries but might not want to start an Android emulator every time.

**Comment by sjelin** [comment]
All comments addressed - again!

**Comment by sjelin** [comment]
But fixed.  Probably not gonna work on windows but honestly I'll do that in a future commit.

**Comment by cnishina** [comment]
@sjelin I'll try it out in Windows. LGTM

**Pull request merged**